hw_9.py

```diff
@@ -0,0 +1,306 @@
+        return None
+
+    def add_material(self, name, details, quantity, availability, price):
+        material = Material(name, details, quantity, availability, price)
+        self.stock.append(material)
+
+    def remove_material(self, material):
+        material = self.find_material(material)
+        if material:
+            self.stock.remove(material)
+
+    def current_quantity(self, material):
+        material = self.find_material(material)
+        if material:
+            print('Current stock of {0} is {1}'.format(material.name, material.quantity))
+        else:
+            print('Material does not exist.')
+
+    def current_quantity_all(self):
+        if self.stock:
+            for material in self.stock:
+                self.current_quantity(material.name)
+        else:
+            print('Warehouse is empty.')
+
+    def create_basket(self):
+        basket = Basket()
+        self.baskets.append(basket)
+        print('Basket with number ', basket.basket_id, ' was created.')
+        return basket
+
+    def find_basket(self, basket_id):
+        if self.baskets:
+            for basket in self.baskets:
+                if basket.basket_id == basket_id:
+                    return basket
+        return None
+
+    def generate_order(self, basket_id):
+        basket = self.find_basket(basket_id)
+        if basket:
+            order = Order()
+            self.orders.append(order)
+            for material in basket.content:
+                #if isinstance(material, Material):
+                    #warehouse_material = self.find_material(material.name)
+                    #print(warehouse_material)
+                if basket.content[material] <= material.quantity:
+                    order.items[material] = basket.content[material]
+                    material.quantity -= basket.content[material]
+                else:
+                    print('Quantity of material {0} is not enough to process your order, so it will not be added.'
+                          .format(material.name))
+            return order
+
+    def find_order(self, order_id):
+        if self.orders:
+            for order in self.orders:
+                if order.order_id == order_id:
+                    return order
+        return None
+
+
+class Category(Warehouse):
+    def __init__(self, name):
+        self.name = name
+        self.members = []
+        super().__init__()
+
+    def add_to_category(self, material):
+        if isinstance(material, Material):
+            self.members.append(material)
+            material.category = self.name
+        else:
+            print('Material does not belong to material class.')
+
+    def print_category(self):
+        id = 0
+        if self.members:
+            for material in self.members:
+                id += 1
+                print(id, 'Material name: ', material.name, 'Material quantity :', material.quantity)
+
+
+basket_id = 0
+
+
+class Basket(Warehouse):
```

Comment:
Basket and Warehouse it's totally different classes. It's like a human and animal, both of them can walk or eat, but they can't be in relation parent and child.